extensions/amp-experiment/0.1/variant.js

```diff
@@ -39,7 +39,7 @@ export function allocateVariant(win, config) {
 
   if (cidScope && config.consentNotificationId) {
     hasConsentPromise = userNotificationManagerFor(win)
-        .then(manager => manager.get(config.consentNotificationId))
+        .then(manager => manager.getNotification(config.consentNotificationId))
```

Comment:
I'm 90% sure this is going to cause problems, even in your example page. The `<amp-user-notification>` may or may not have been parsed yet into DOM, we have no way of really knowing. Then we're going to assert that it's there in the next promise block.

Yes, waiting for the notification to register itself is "blocking", but it'll register itself before the page is un-hidden, so it's not aggressively blocking. I don't think waiting is an issue.